### ISSUE TITLE
yolov4 not detecting properly

### ISSUE THREAD

**Issue by Akotkar23** [AlexeyAB/darknet#7708]
@AlexeyAB I am trying to detect small objects using YOLOv4.
Done the necessary changes in config file and able to initiate the training but after training getting this kind of results. What can be the issue. 
Attaching the image.
![download (7)](https://user-images.githubusercontent.com/68338164/118227058-880ba480-b4a5-11eb-84a5-a949620e4ee4.png)
Also, for some images getting the following issue "STB Reason: can't fopen"
![Untitled](https://user-images.githubusercontent.com/68338164/118228485-e3d72d00-b4a7-11eb-8477-a02fd073f25c.png)
Thank You!

**Comment by stephanecharette** [comment]
What did you set max batches to when training?
Have you looked at the FAQ?  https://www.ccoderun.ca/programming/darknet_faq/#tldr

**Comment by Akotkar23** [comment]
> What did you set max batches to when training?
> Have you looked at the FAQ? https://www.ccoderun.ca/programming/darknet_faq/#tldr

Yes, I've checked FAQ.
max batches = 6000

**Comment by stephanecharette** [comment]
With max batches set to 6000, I assume you only have 1 class and a few hundred images?

What did you chart.png look like?

**Comment by Akotkar23** [comment]
> With max batches set to 6000, I assume you only have 1 class and a few hundred images?
> 
> What did your chart.png look like?

Yes, I have 1 class with less than 100 images. Also objects are very small.
Didn't got any graph.

**Comment by stephanecharette** [comment]
1) You probably don't have enough images:  https://www.ccoderun.ca/programming/darknet_faq/#how_many_images
2) You probably don't have the right repo checked out since you don't have a chart.png:  https://www.ccoderun.ca/programming/darknet_faq/#tldr
3) If your images are so small that darknet cannot train with them, you probably need to turn on tiling for both training and inference: https://www.ccoderun.ca/programming/darknet_faq/#tldr

That last one is easy to tell.  You have to answer 3 questions:
1) What size is your network?  (At the top of your .cfg is a width and height.)
2) What size are your images?
3) What size are your objects within those images?

**Comment by AlexeyAB** [comment]
Type 2 mAP values - what accuracy can you get on Training and Validation datasets? https://github.com/AlexeyAB/darknet/wiki/FAQ---frequently-asked-questions#1-i-get-low-accuracy

**Comment by Akotkar23** [comment]
> 1. You probably don't have enough images:  https://www.ccoderun.ca/programming/darknet_faq/#how_many_images
> 2. You probably don't have the right repo checked out since you don't have a chart.png:  https://www.ccoderun.ca/programming/darknet_faq/#tldr
> 3. If your images are so small that darknet cannot train with them, you probably need to turn on tiling for both training and inference: https://www.ccoderun.ca/programming/darknet_faq/#tldr
> 
> That last one is easy to tell. You have to answer 3 questions:
> 
> 1. What size is your network?  (At the top of your .cfg is a width and height.)
> 2. What size are your images?
> 3. What size are your objects within those images?

1. H=416, W=416
2. Size of image  = 4000*3000
3. objects are very small. Attached the reference image. I have labelled the object with red square. PFA.
![reference](https://user-images.githubusercontent.com/68338164/118595434-b0a6dd80-b7c8-11eb-8acf-f9585627253c.png)

**Comment by stephanecharette** [comment]
Once you resize your 4000x3000 images to 416x416, that red box you have is at the edge of being detectable.

- You're certain you've annotated all of these objects in all images, without having missed any?
- Have you checked which repo you checked out? Because if you don't have chart.png after training, it normally means you're using the wrong one which was abandoned many years ago.
- Have you checked your annotations with a tool like DarkMark?
- Have you considered using image tiling?
- Have you considered increasing the network dimensions?

**Comment by Akotkar23** [comment]
> Once you resize your 4000x3000 images to 416x416, that red box you have is at the edge of being detectable.
> 
> * You're certain you've annotated all of these objects in all images, without having missed any?
> * Have you checked which repo you checked out? Because if you don't have chart.png after training, it normally means you're using the wrong one which was abandoned many years ago.
> * Have you checked your annotations with a tool like DarkMark?
> * Have you considered using image tiling?
> * Have you considered increasing the network dimensions?

I have labelled all the objects in all the images.
Thanks for the suggestions I will look into it as I did not considered using image tiling and increasing the network dimensions.
